src/connector/src/sink/kafka.rs

```diff
@@ -344,8 +345,20 @@ fn datum_to_json_object(field: &Field, datum: DatumRef<'_>) -> ArrayResult<Value
             // fixme
             json!(v.to_string())
         }
-        (DataType::Time, ScalarRefImpl::NaiveTime(_v)) => {
-            unimplemented!()
+        (DataType::Date, ScalarRefImpl::NaiveDate(v)) => {
+            json!(v.to_text())
+        }
+        (DataType::Time, ScalarRefImpl::NaiveTime(v)) => {
+            json!(v.to_text())
+        }
+        (DataType::TIMESTAMP, ScalarRefImpl::NaiveDateTime(v)) => {
+            json!(v.to_text())
+        }
+        (DataType::TIMESTAMPZ, ScalarRefImpl::Int64(v)) => {
+            json!(v)
```

Comment:
* If an output like `"v1": "2022-12-01 12:00:00+00:00",` is preferred, use the new `v.to_text_with_type`.
  * Given all 5 chrono types are using string, it can be shorten to a single match arm `(DataType::Date | DataType::Time | ..., _) => json!(v.to_text_with_type(data_type))`
* If an output like `"v1": 1669896000000000` is preferred, keep `v` as it is right now. Note this is in microseconds.